src/Options.js

```diff
@@ -250,10 +257,38 @@ export class Options {
 };
 
 
-// A distinguish instance shared internally via module
-//
-export var options = new Options();
+class ParseOptions {
+  constructor(options) {
+    this.proxiedOptions_ = options;
 
+    Object.keys(featureOptions).forEach((name) => {
```

Comment:
done.  I also re-order the file, so the diff will be larger than one might expect.